repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,227 @@
+        'innodb-concurrency-tickets', 'innodb-log-file-size',
+        'innodb-old-blocks-time', 'innodb-open-files', 
+        'innodb-purge-batch-size', 'innodb-undo-logs',
+        'max-connect-errors', 'max-relay-log-size', 'myisam-sort-buffer-size',
+        'optimizer-switch'
+    ]
+    options_renamed = {
+        'engine-condition-pushdown': 'set optimizer_switch="engine_condition_pushdown=on"',
+        'innodb-fast-checksum': 'innodb-checksum-algorithm',
+        'innodb-flush-neighbor-pages': 'innodb-flush-neighbors',
+        'innodb-stats-auto-update': 'innodb-stats-auto-recalc'
+    }
+    options_not_in8 = [
+        'innodb-adaptive-flushing-method', 'innodb-autoextend-increment',
+        'innodb-blocking-buffer-pool-restore', 'innodb-buffer-pool-pages',
+        'innodb-buffer-pool-pages-blob', 'innodb-buffer-pool-pages-index',
+        'innodb-buffer-pool-restore-at-startup', 'innodb-buffer-pool-shm-checksum',
+        'innodb-buffer-pool-shm-key', 'innodb-checkpoint-age-target',
+        'innodb-dict-size-limit', 'innodb-doublewrite-file',
+        'innodb-ibuf-accel-rate', 'innodb-ibuf-active-contract',
+        'innodb-ibuf-max-size', 'innodb-import-table-from-xtrabackup',
+        'innodb-index-stats', 'innodb-lazy-drop-table', 
+        'innodb-merge-sort-block-size', 'innodb-persistent-stats-root-page',
+        'innodb-read-ahead', 'innodb-recovery-stats',
+        'innodb-recovery-update-relay-log', 'innodb-stats-update-need-lock',
+        'innodb-sys-stats', 'innodb-table-stats', 'innodb-thread-concurrency-timer-based',
+        'innodb-use-sys-stats-table', 'xtradb-admin-command'
+    ]
+    
+    warn_changed = []
+    warn_renamed = {}
+    warn_removed = []
+
+    # TODO this could check also values
+    for changed_opt in options_changed_def:
+        for used_opt in used_opts:
+            if re.search(r"" + changed_opt + "[\s=]{1}", used_opt):
+                warn_changed.append(changed_opt)
+
+    for renamed_opt in options_renamed:    
+        for used_opt in used_opts:
+            if re.search(r"" + renamed_opt + "[\s=]{1}", used_opt):
+                warn_renamed[renamed_opt] = options_renamed[renamed_opt]
+
+    for removed_opt in options_not_in8:
+        for used_opt in used_opts:
+            if re.search(r"" + removed_opt + "[\s=]{1}", used_opt):
+                warn_removed.append(removed_opt)
+
+    return (warn_changed, warn_renamed, warn_removed)
+
+
+def get_mariadb_packages():
+    """ Get list of installed MariaDB packages """
+
+    for rpm_pkgs in api.consume(InstalledRedHatSignedRPM):
+        return [pkg for pkg in rpm_pkgs.items if pkg.name.startswith('mariadb')]
+
+
+def generate_report(packages):
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = 'https://access.redhat.com/articles/4055661'
+
+    config = get_configs_as_text(get_config_files("/etc/my.cnf"))
```

Comment:
Nit: Move /etc/my.cnf to a constant? Or a local report-specific var like DOC_URL at least?